internal/analytics/analytics.go

```diff
@@ -206,6 +208,15 @@ func sendEventAndLog(category, action, label string, dimensions map[string]strin
 }
 
 func sendEvent(category, action, label string, dimensions map[string]string) error {
+	err := sendGAEvent(category, action, label, dimensions)
```

Comment:
I think that `sendGAEvent` should only be [this part](https://github.com/ActiveState/cli/pull/1222/files#diff-895f0dd442920ec2aa8a34401e292e8d73c29054d75da4a2541f10380e901eafL223-L238), and the handling of deferred analytics event should be handled by both the S3 Pixel and GA ie., in this method.

Right now, the deferred events are always sent to S3, even when they should be deferred.

Reply:
No wait!  The GA Events are also sent! 
@Naatan Shouldn't there be a `return nil` after line 220, as we do *not* want to send anything if `deferAnalytics` is `true`?

If I understand this correctly, we are still sending a GA event with every `state shim` command immediately, but *also* as a deferred bulk operation later...

Reply:
I think you're right.. we need to actually cancel out when deferring.